cpp/src/request_tag.cpp

```diff
@@ -70,8 +70,6 @@ void RequestTag::callback(void* request, ucs_status_t status, const ucp_tag_recv
     std::snprintf(_status_msg.data(), _status_msg.size(), fmt, info->length, _length);
   }
 
-  _status = status;
-
   Request::callback(request, status);
```

Comment:
So we're just deferring to the superclass to set the status now?

Reply:
Correct, I've commented in the question above regarding the reasons.